code/modules/reagents/chemistry/machinery/pandemic.dm

```diff
@@ -172,6 +177,46 @@
 	add_fingerprint(usr)
 	return
 
+//Prints a nice virus release form. Props to Urbanliner for the layout
+/obj/machinery/computer/pandemic/proc/print_form(var/datum/disease/advance/D, mob/living/user)
+	D = archive_diseases[D.GetDiseaseID()]
+	if(!(printing) && D)
+		var/reason = input(user,"Enter a reason for the release", "Write", null) as message
+		reason += "<span class=\"paper_field\"></span>"
+		var/english_symptoms = list()
+		for(var/datum/symptom/S in D.symptoms)
```

Comment:
Assuming you know that D.symptoms only contains objects of type `/datum/symptom` you can use a typeless loop here which is faster.

```
for(var/I in D.symptoms)
    var/datum/symptom/D = I
   ...etc()
```

Reply:
Nice!
Didn't know it would be faster that way

Reply:
It's faster because you avoid an `istype()` check altogether.